src/main/java/org/jabref/logic/importer/util/GrobidService.java

```diff
@@ -84,6 +84,25 @@ public List<BibEntry> processPDF(Path filePath, ImportFormatPreferences importFo
 
         String httpResponse = response.body();
 
+        return getBibEntries(importFormatPreferences, httpResponse);
+    }
+
+    public List<BibEntry> processReferences(Path filePath, ImportFormatPreferences importFormatPreferences) throws IOException, ParseException {
+        Connection.Response response = Jsoup.connect(grobidPreferences.getGrobidURL() + "/api/processReferences")
```

Comment:
There should be a preference to offer the choice between

* use Grobid online service
* use local PDF parsing

The feature request #10200 had the second one in mind. The data of the PDF should not be shared with any online service. Reason: The PDF is under review with the constraint that the PDF must not be shared with third parties.

Reply:
For Grobid, would this preference be the same as what we already have under `Import and Export -> Remote Services`?
For example, if the user has not allowed sending data to Grobid in the 'Remote Services' preferences, should we just use the local PDF parsing option by default?
And vice-versa, if the user has enabled sending data to Grobid, we would just use Grobid since the parsed result would be of higher quality

Reply:
I think it's fine to implement a remote parsing first (not many people have a local grobid running, nor the strict privacy restrictions that olly mentioned above). Yes, reusing the same toggle sounds like a good idea.

Reply:
Grobid is up and running again at http://grobid.jabref.org:8070

Reply:
For the local only parsing option, is it to be done assuming a local Grobid server is running? or does it require writing a custom PDF reference parser using PDFBox for example

Reply:
I would say for now go with a hard-coded static grobid url. For testing purposes (as a dev) one can then easily change it to target a local grobid server. Once the feature works and is tested by a few users, we can explore what config options are good to have (e.g. local/own grobid server, or a custom PDF parser).

Reply:
@tobiasdiez Maybe you overlooked the quoted code.

I highlight ´grobidPreferences.getGrobidURL() + "/api/processCitation"`  in the screenshot.

![image](https://github.com/JabRef/jabref/assets/1366654/a9f1a02e-54d8-49a1-9791-0b7fb2a44fb3)

Thus, no need for hard-coded URLs.

The preferences even allow for searching the preference:

![image](https://github.com/JabRef/jabref/assets/1366654/704f5006-220a-49bd-8623-949aa0f578d5)

Reply:
> There should be a preference to offer the choice between
> 
>     * use Grobid online service
>     * use local PDF parsing

With "local PDF parsing" I meant `org.jabref.logic.importer.fileformat.PdfContentImporter` which does NOT use GROBID at all.

I meant: The user does not install GROBID locally, but uses some custom parsing logic, which works 99,999% reliably for IEEE papers.  Having a guess for solving https://github.com/JabRef/jabref/issues/10200#issuecomment-1707219332 is not acceptable as paper acceptance will rely on that feature.

It is fine for me if GROBID achieves a high reliability. Let's see, how this PR will continue.

It is also OK for me to leave the local heuristics for future work. I just wanted to state that GROBID might not be as reliable as desired.

Reply:
Thanks for adding the PDF sample! I'll take a look at PdfContentImporter to see what additional parsing is required for the references